diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54ab9ea..628b8c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -146,9 +146,9 @@ add_subdirectory(synchronizer)
 add_subdirectory(examples)
 
 # some tests
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 # cli
-add_subdirectory(sinksh)
+#add_subdirectory(sinksh)
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff --git a/common/domain/applicationdomaintype.h b/common/domain/applicationdomaintype.h
index c931e91..c05fab2 100644
--- a/common/domain/applicationdomaintype.h
+++ b/common/domain/applicationdomaintype.h
@@ -74,7 +74,7 @@
 
 #define SINK_INDEX_PROPERTY(TYPE, NAME, LOWERCASENAME) \
     struct NAME { \
-        static constexpr const char *name = #LOWERCASENAME; \
+        SINK_EXPORT static constexpr const char *name = #LOWERCASENAME; \
         typedef TYPE Type; \
     }; \
 
diff --git a/examples/caldavresource/CMakeLists.txt b/examples/caldavresource/CMakeLists.txt
index 314a8f5..8a02197 100644
--- a/examples/caldavresource/CMakeLists.txt
+++ b/examples/caldavresource/CMakeLists.txt
@@ -12,4 +12,4 @@ target_link_libraries(${PROJECT_NAME} sink_webdav_common sink Qt5::Core Qt5::Net
 
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${SINK_RESOURCE_PLUGINS_PATH} RUNTIME DESTINATION ${SINK_RESOURCE_PLUGINS_PATH})
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
diff --git a/examples/carddavresource/CMakeLists.txt b/examples/carddavresource/CMakeLists.txt
index cca7ea7..986b112 100644
--- a/examples/carddavresource/CMakeLists.txt
+++ b/examples/carddavresource/CMakeLists.txt
@@ -10,4 +10,4 @@ target_link_libraries(${PROJECT_NAME} sink_webdav_common sink Qt5::Core Qt5::Net
 
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${SINK_RESOURCE_PLUGINS_PATH} RUNTIME DESTINATION ${SINK_RESOURCE_PLUGINS_PATH})
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
diff --git a/examples/imapresource/CMakeLists.txt b/examples/imapresource/CMakeLists.txt
index 290a487..d214e9b 100644
--- a/examples/imapresource/CMakeLists.txt
+++ b/examples/imapresource/CMakeLists.txt
@@ -13,4 +13,4 @@ target_link_libraries(${PROJECT_NAME} sink Qt5::Core Qt5::Network KF5::Mime KIMA
 
 install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${SINK_RESOURCE_PLUGINS_PATH} RUNTIME DESTINATION ${SINK_RESOURCE_PLUGINS_PATH})
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
diff --git a/examples/maildirresource/CMakeLists.txt b/examples/maildirresource/CMakeLists.txt
index b1c6489..6738850 100644
--- a/examples/maildirresource/CMakeLists.txt
+++ b/examples/maildirresource/CMakeLists.txt
@@ -12,4 +12,4 @@ install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION ${SINK_RESOURCE_PLUGINS_PATH
 
 add_definitions(-DTESTDATAPATH="${CMAKE_CURRENT_SOURCE_DIR}/tests/data")
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
diff --git a/examples/mailtransportresource/CMakeLists.txt b/examples/mailtransportresource/CMakeLists.txt
index 56ba9e7..c54b06c 100644
--- a/examples/mailtransportresource/CMakeLists.txt
+++ b/examples/mailtransportresource/CMakeLists.txt
@@ -29,4 +29,4 @@ target_link_libraries(sink_smtp_test
 )
 install(TARGETS sink_smtp_test ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
 
-add_subdirectory(tests)
+#add_subdirectory(tests)
