diff -Nrub -x '*~' -x '*\.rej' -x '*\.orig' -x'*\.o' -x '*\.pyc' xapian-core-1.4.5.orig/Makefile.in xapian-core-1.4.5/Makefile.in
--- xapian-core-1.4.5.orig/Makefile.in	2017-10-15 19:32:45.000000000 -0700
+++ xapian-core-1.4.5/Makefile.in	2018-03-20 01:27:25.000000000 -0700
@@ -3733,11 +3733,11 @@
 install-exec-local: xapian-config
 	$(MKDIR_P) "$(DESTDIR)$(bindir)"
 	$(SED) '/##UNINSTONLYBEGIN/,/##UNINSTONLYEND/d' xapian-config > tmp.xc
-	$(INSTALL_SCRIPT) tmp.xc "$(DESTDIR)$(bindir)$$dir/`echo xapian-config|$(SED) -e '$(transform)'`"
+	$(INSTALL_SCRIPT) tmp.xc "$(DESTDIR)$(bindir)/xapian-config"
 	rm -f tmp.xc
 
 uninstall-local:
-	rm -f "$(DESTDIR)$(bindir)$$dir/xapian-config"
+	rm -f "$(DESTDIR)$(bindir)/xapian-config"
 docsource.mk: preautoreconf
 	cd $(top_srcdir) && ./preautoreconf
 
